### ISSUE TITLE
Github releases is blocked in many corporations

### ISSUE THREAD

**Issue by gilescope** [rust-analyzer/rust-analyzer#3103]
but vscode extensions are not loaded through the standard vscode api. For those that live behind the great firewall of corporation, is there a way we can make their lives easier without many engineering teams trying to figure out separate ways to solve the problem?

Two solutions I can see that might work:

Create platform specific extensions that are bundled with the typescript. (The typescript isn't the big bit)

Create platform specific extensions along with the current extension and then ask VS code to install the appropriate additional extension. Ideally VSCode should have a general solution to this problem as codelldb has exactly the same issue.

I have a vision of a future where things 'just work' even at work.

**Comment by gilescope** [comment]
The right answer probably depend on VsCode/Marketplace providing a nice mechanism: https://github.com/microsoft/vscode/issues/23251
(Though there are suggestions on that thread as to how an extensions can install a different extension)

**Comment by Veetaha** [comment]
VSCode `cpptools` extension [bundles their binaries in platform-specific `.vsix` packages](https://github.com/microsoft/vscode-cpptools/releases) in their releases. And they do use github api to download updates. I am not sure what they actually publish to marketplace, this needs more thorough investigation.

**Comment by gilescope** [comment]
Interestingly the current mechanism works well in nixos (which surprises me greatly). I guess it's downloading to somewhere that is read/write-able. (Well done)

**Comment by Veetaha** [comment]
We are downloading to [`globalStoragePath`](https://code.visualstudio.com/api/extension-capabilities/common-capabilities#data-storage). Where would you assume we might do something not allowed in companies? I guess Github Releases shouldn't be blocked, it's just a source to download files that is one of trillions, or do you say it might be blacklisted?

**Comment by Veetaha** [comment]
And, yeah, the documentation says that globalStorage is read/writable ))

**Comment by Veetaha** [comment]
If this was your main concern @gilescope , then, should we close this issue, or do you think there might be something else? I also noted that long-runing vscode issue about platform-specific extensions, thanks for the pointer!

**Comment by lnicola** [comment]
`globalStoragePath` is `~/.config/Code/User/globalStorage`, so it's writable by the user running Code. You can use this to side-load (via a flash drive or something) the LSP binary in case downloading it automatically doesn't work.

**Comment by gilescope** [comment]
Where it’s being stored is great for me on nixos - I might try and persuade codelldb to do the same. For work, I guess it’s about minimising dependencies - at the moment the vscode marketplace needs to be unblocked and amazon s3 needs to be unblocked (as this is where github releases are stored). It’s a lot easier to justify a hole to vscode marketplace than a hole to some s3 bucket. Maybe close this ticket and see if others reopen it.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by kiljacken** [comment]
If your workplace blocks both vscode marketplace and S3, we really have no realistic way to get a hole through. That's about as legitimate as it gets in my mind. Building from source or smuggling it on a usb stick seems the only remaining solutions more portable.

**Comment by matklad** [comment]
I am going to close this as explicitely out of scope: we don't have resources to invest into strategies more elaborate than github releases. If this is an acute problem, anyone is welcome to mirror the binaries in an accessible location!

**Comment by prashantvc** [comment]
👋 I am Program manager on VS Marketplace team. We are working on solving this issue of distributing dependencies from the Marketplace. I would love to get your opinion (30mins call), it will help us to build a better solution

Please use the link to pick time that works for you https://aka.ms/vsm-call

**Comment by matklad** [comment]
Thanks, booked a slot for tomorrow!
